pandas/_libs/tslibs/tzconversion.pxd

```diff
@@ -1,9 +1,22 @@
 from cpython.datetime cimport tzinfo
-from numpy cimport int64_t
+from numpy cimport int64_t, intp_t, ndarray
 
 
 cdef int64_t tz_convert_utc_to_tzlocal(int64_t utc_val, tzinfo tz, bint* fold=*)
 cpdef int64_t tz_convert_single(int64_t val, tzinfo tz1, tzinfo tz2)
 cdef int64_t tz_localize_to_utc_single(
     int64_t val, tzinfo tz, object ambiguous=*, object nonexistent=*
 ) except? -1
+
+
+cdef class Localizer:
+    cdef:
+        bint use_utc, use_tzlocal, use_fixed, use_pytz
+        int noffsets
+        int64_t* utcoffsets
+        intp_t* positions
+        ndarray positions_arr  # needed to avoid segfault
```

Comment:
Do we even need both positions_arr and positions or can we just use the former?

Reply:
we _could_ just use the former, but i think we get a perf boost from indexing on the latter

Reply:
If you can see a difference then sure, but if not would definitely be cleaner to just stick with the ndarray